File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int cntNeg = 0;
        for (int num : nums) {
            if (num == 0) {
                return 0;
            }
            cntNeg += num < 0;
        }
        return cntNeg % 2 == 0 ? 1 : -1;
    }
};

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        cntNeg = 0
        
        for num in nums:
            if num == 0:
                return 0
            cntNeg += int(num < 0)
        
        return 1 if cntNeg % 2 == 0 else -1